### PR TITLE
GH-36: Implementing the Profile Retrieval feature and Unit tests

### PR DESCRIPTION
**Pull request by Ziadsharkos**

### CODE REVIEW

**Review by KhalilGarali — APPROVED** [review]
Looks good!

File: Microservices/user-service/src/test/java/app/sportahub/userservice/controller/UserControllerTest.java [hunk 1]
```diff
@@ -22,14 +24,16 @@
...
 import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.when;
 import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
 
-@ActiveProfiles("user-service.test")
+import static org.hamcrest.Matchers.*;
+
 @WebMvcTest(UserController.class)
 @Import(TestSecurityConfig.class)
 public class UserControllerTest {
```
Comment (walidoow) [inline Microservices/user-service/src/test/java/app/sportahub/userservice/controller/UserControllerTest.java:39]: The tests of all of this class fail. @Ziadsharkos it's ok, simply remove this test file from the PR, and you will add the tests on our next iteration 🔄

**Review by walidoow — CHANGES_REQUESTED** [review]

File: Microservices/user-service/src/test/java/app/sportahub/userservice/controller/UserControllerTest.java (hunk 1 above)
Comment (Ziadsharkos) [reply Microservices/user-service/src/test/java/app/sportahub/userservice/controller/UserControllerTest.java]: Agreed!

**Review by walidoow — APPROVED** [review]
Great work, thank you! 🎉

**Pull request merged**